[Ong Kim Lai] iP

ongkimlai opened the pull request:
# HI THIS IS BOBBY 
Bobby will be your personal task manager, managing your deadlines!

Bobby gives you some peace of mind! It is:
* Lightweight
* Easy to use
* **VERY** HELPFUL 😀 
* _FREE_

Start using Bobby today! 

All you need to do is  3 simple steps: 
1. Download the jar file [here](https://github.com/ongkimlai/ip/releases/download/A-Jar/ip.jar) 
2. Open Terminal in the same directory as ip.jar 
3. Run the command `java -jar ip.jar` 
And you are done! Bobby has now started and you can start using Bobby now. 

Features:
- [ ] Keeping a To-Do List
- [ ] Managing deadlines tasks
- [ ] Managing events tasks

If you are a Java programmer and want to practice some Java, here is the ``main`` class: 
```ruby
public static void main(String[] args) {
    new Bobby(FILE_LOCATION).run();
}
```

Commit:
Add Gradle support

ongkimlai committed:
level 1

ongkimlai committed:
changed name to Bobby

ongkimlai committed:
lvl 2

ongkimlai committed:
lvl 3

ongkimlai committed:
changed to Bobby.java

ongkimlai committed:
minor changes to texts, added Deadline.java

ongkimlai committed:
lvl 4

ongkimlai committed:
A-TextUITesting

ongkimlai committed:
lvl5

ongkimlai committed:
Complete level-5.

ongkimlai committed:
added already_mark exceptions

ongkimlai committed:
A-Enums

ongkimlai committed:
Modify style

ongkimlai committed:
Level-7

ongkimlai committed:
Complete Level-8

ongkimlai committed:
Merge branch 'branch-Level-7'

ongkimlai committed:
Merge branch-Level-8

ongkimlai committed:
Implement more OOP

ongkimlai committed:
Implement Packages

ongkimlai committed:
A-JUnit

ongkimlai committed:
A-Jar

ongkimlai committed:
A-Javadoc

ongkimlai committed:
A-CodingStandard

ongkimlai committed:
Implement Level-9: Find

ongkimlai committed:
Merge branch 'branch-A-JavaDoc'

ongkimlai committed:
Merge branch 'branch-A-CodingStandard'

ongkimlai committed:
Merge branch 'branch-Level-9'

ongkimlai committed:
Merge branch 'add-gradle-support'

ongkimlai committed:
Add gradle support

ongkimlai committed:
Implement checkstyle

Punpun1643 commented on src/main/java/bobby/Ui.java:
```diff
@@ -0,0 +1,138 @@
+        this.sc = sc;
+    }
+
+    public void print(String message) {
```
Don't forget that you have to write header comments for all classes, public methods except for getter/setter.

Punpun1643 commented on src/main/java/bobby/Storage.java:
```diff
@@ -0,0 +1,48 @@
+
+    @SuppressWarnings("unchecked")
+    public ArrayList<Task> loadTasks() throws StorageException, FileNotFoundException {
+        ArrayList<Task> task;
```
Remember that plural form should be used on names representing a collection of objects.

Punpun1643 commented on src/main/java/bobby/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,86 @@
+package bobby.command;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import bobby.Storage;
+import bobby.Ui;
+import bobby.exception.BobbyException;
+import bobby.exception.DeadlineException;
+import bobby.task.Deadline;
+import bobby.task.TaskList;
```
I like how the import ordering is really clean and I also like how import statements are separated based on category 👍

shurvirarora replied:
Agreed. Organised import statements 👍

Punpun1643 commented on src/main/java/bobby/command/ToDoCommand.java:
```diff
@@ -0,0 +1,62 @@
+    /** The full user input command */
+    private String fullCommand;
+    /** The full user input command in array form */
+    private String[] fullCommandArr;
```
I think it might be better to name it in a plural form such as fullUserCommands 😄

Punpun1643 commented on src/main/java/bobby/Parser.java:
```diff
@@ -0,0 +1,43 @@
+    public static Command parse(String fullCommand) {
+        String[] fullCommandArr = fullCommand.split(" ");
+        Command command;
+        switch(fullCommandArr[0].toLowerCase()) {
```
Really good in remembering to follow the module's switch statement standard...same goes for other switch statements. 👍

Punpun1643 reviewed (commented):
Generally, the code looks really clean. 👍  Just remember the header comments for public methods.

shurvirarora commented on src/main/java/bobby/command/ByeCommand.java:
```diff
@@ -0,0 +1,40 @@
+    /**
+     * Creates a ByeCommand Object
+     */
+    public ByeCommand(){
```
Missed out a white space between '()' and '{'

shurvirarora commented on src/main/java/bobby/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,86 @@
+            throw new DeadlineException("no_slash");
+        } else if (fullCommand.substring(fullCommand.indexOf("/") + 1).isBlank()) { // nothing after time
+            throw new DeadlineException("no_date");
+        } else if (!isValidDate(fullCommand.substring(fullCommand.length() - 10))) { // invalid date
```
For long lines, perhaps including a line break before the operator could make it look neater 👍

shurvirarora commented on src/main/java/bobby/task/Task.java:
```diff
@@ -0,0 +1,65 @@
+
+public class Task implements Serializable, Comparable<Task> {
+    private String taskName;
+    private boolean isDone;
```
Good naming for boolean with use of "is" 👍

shurvirarora reviewed (commented):
Overall code LGTM! Just some nits to fix (e.g. white spacing and line breaks for long lines)

ongkimlai committed:
Add JavaFX GUI, listing feature buggy

ongkimlai committed:
Fix vbox bug and change pictures

ongkimlai committed:
Add background colour

Shivang211 commented on src/main/java/bobby/Parser.java:
```diff
@@ -0,0 +1,43 @@
+        String[] fullCommandArr = fullCommand.split(" ");
+        Command command;
+        switch(fullCommandArr[0].toLowerCase()) {
+        case "bye": command = new ByeCommand();
```
There needs to be a \n after case according to the Java coding standards for this module

Shivang211 commented on src/main/java/bobby/command/EventCommand.java:
```diff
@@ -0,0 +1,89 @@
+        } else if (!isValidDate(fullCommand.substring(fullCommand.length() - 10))) { // invalid date
+            throw new EventException("invalid_date");
+        }
+        Event newEvent = new Event(fullCommand.substring(fullCommand.indexOf(" ") + 1,
```
Perhaps can give an extra empty line in long stretches of code for better readability

Shivang211 commented on src/main/java/bobby/task/Event.java:
```diff
@@ -0,0 +1,23 @@
+    private String at;
+    private LocalDate date;
+
+    public Event(String taskName, String at) {
```
Can put a better variable name than 'at' to explain the value it holds better, like time etc.

Shivang211 commented on src/main/java/bobby/exception/FindException.java:
```diff
@@ -0,0 +1,23 @@
+            break;
+        default: errorString = "\tCannot find any task matching the keyword.";
+        }
+        return errorString;
```
Naming for errors is nicely done, explains the data the variable is supposed to hold 👍

Shivang211 reviewed (commented):
Well done!! 👍

ongkimlai committed:
branch-Level-10

ongkimlai committed:
Fix delete and unmark bug

ongkimlai committed:
Implement assertions into Command classes

ongkimlai committed:
Change Parser to dissect the user input instead of Command classes

Command class should only care about executing, not interpreting command

Allows for clearer and cleaner abstraction

Parser class is meant to read the user command and check for errors, then pass calls the correct Command method.

ongkimlai committed:
Merge pull request #1 from ongkimlai/branch-A-CodeQuality

Branch A-CodeQuality

ongkimlai committed:
Merge branch 'master' into branch-A-Assertions

ongkimlai committed:
Merge pull request #2 from ongkimlai/branch-A-Assertions

Implement assertions into Command classes

ongkimlai committed:
Add sorting functionality to the 'list' command

Allows users to easily see and remind them what deadlines and events they have that requires their attention more.

Made the Task class implement Comparable interface to allow for Java's sort method.

ongkimlai committed:
Improve GUI and added more javadocs

ongkimlai committed:
Add a brief user guide

ongkimlai committed:
Set theme jekyll-theme-tactile

ongkimlai committed:
Fix minor typos

ongkimlai committed:
Fix minor typos again

ongkimlai committed:
Implement github page files

ongkimlai committed:
Set theme jekyll-theme-slate

ongkimlai committed:
Set theme jekyll-theme-midnight

ongkimlai committed:
Set theme jekyll-theme-architect

ongkimlai committed:
Add CI

ongkimlai committed:
Merge branch 'master' of https://github.com/ongkimlai/ip

ongkimlai committed:
Remove logo string and cleaned up code

ongkimlai committed:
Bug fix
